src/gebt_poc/solver.cpp

```diff
@@ -468,28 +468,28 @@ void ConstraintsResidualVector(
 ) {
     auto translation_0 = Kokkos::subview(gen_coords, Kokkos::make_pair(0, 3));
     auto rotation_0 = Kokkos::subview(gen_coords, Kokkos::make_pair(3, 7));
-    auto rotation_matrix_0 = gen_alpha_solver::EulerParameterToRotationMatrix(rotation_0);
-    auto position_0 = Kokkos::subview(position_vector, Kokkos::make_pair(0, 3));
 
-    // position = position_0 + translation_0
-    auto position = Kokkos::View<double[kNumberOfVectorComponents]>("position");
-    Kokkos::deep_copy(position, position_0);
-    KokkosBlas::axpy(1., translation_0, position);
-
-    // rotated_position = rotation_matrix_0 * position
-    auto rotated_position = Kokkos::View<double[kNumberOfVectorComponents]>("rotated_position");
-    KokkosBlas::gemv("N", 1., rotation_matrix_0, position, 0., rotated_position);
-
-    // Assemble the constraint residual vector
-    // {constraints_residual}_6x1 = {
-    //    {translation_0}_3x1
-    //    {rotated_position - position_0}_3x1
-    // }
-    auto constraints_residual_1 = Kokkos::subview(constraints_residual, Kokkos::make_pair(0, 3));
-    Kokkos::deep_copy(constraints_residual_1, translation_0);
-    auto constraints_residual_2 = Kokkos::subview(constraints_residual, Kokkos::make_pair(3, 6));
-    Kokkos::deep_copy(constraints_residual_2, rotated_position);
-    KokkosBlas::axpy(-1., position_0, constraints_residual_2);
+    // For the GEBT, proof of concept problem, the target translation and rotation is zero,
+    // so the residual vector is based on the generalized coordinates at the first node
```

Comment:
Apologies for being nitpicky here - would the following make things a little more clear?

```
// For the GEBT proof of concept problem (i.e. the clamped beam), the dofs are enforced to be zero at
// the left end of the beam, so the constraint residual is simply based on the generalized
// coordinates at the first node
```

Reply:
Updated